Reduce KafkaSender requirements for replica acknowledgements

**Describe the bug**
KafkaSender (Sarama client) in KafkaSource upgrade tests requires confirmation from all nodes. This is part of the wathola-sender in upgrade tests and sending regular "Step" events can take too long. This might prevent the infinite loop in wathola-sender from handling the SIGTERM, resulting in not sending a "Finished" event. It is not required to get confirmation from all nodes and it should make the call exit more quickly - making the upgrade tests more stable.

The error from upgrade tests looks like this:
```
verify.go:52: Error fetching complete/inactive report: timed out waiting for the condition
        Report: &{State:active EventsSent:2502 TotalRequests:0 Thrown:{Duplicated:[] Missing:[] Unexpected:[] Unavailable:[]}} 
```

 **Expected behavior**
Relax the required acks in Sarama sender to block the sender for a shorter time.

 **To Reproduce**
This is hardly reproducible but it can be done by running the KafkaSource continual upgrade tests from this repo. When there are other tests utilizing the Kafka cluster and put it under load, some requests might take longer.

 **Knative release version**

 **Additional context**
Add any other context about the problem here such as proposed priority

--- a/test/upgrade/continual/kafkasender.go
+++ b/test/upgrade/continual/kafkasender.go
@@ -74,6 +74,8 @@ func (k *kafkaSender) SendEvent(ce cloudevents.Event, rawEndpoint interface{}) e
 		WithClientId("continualtests-kafka-sender").
 		WithDefaults().
 		Build(k.ctx)
+	// Prevent excessive wait time.
+	conf.Producer.RequiredAcks = sarama.WaitForLocal
 	if err != nil {
 		return err
 	}

